sprite.cpp

```diff
@@ -67,13 +67,96 @@ void Sprite::transform() {
+				&& abs(get<X>(current_target) - get<X>(last_target)) < abs((get<X>(current_target) + horizontal_boundary * 2) - get<X>(last_target))) {
+				get<X>(new_target) += get<X>(current_target) - get<X>(last_target);
+			} else if (abs((get<X>(current_target) - horizontal_boundary * 2) - get<X>(last_target)) < abs((get<X>(current_target) + horizontal_boundary * 2) - get<X>(last_target))) {
+				get<X>(new_target) += (get<X>(current_target) - horizontal_boundary * 2) - get<X>(last_target);
+			} else {
+				get<X>(new_target) += (get<X>(current_target) + horizontal_boundary * 2) - get<X>(last_target);
+			}
+			if (abs(get<Y>(current_target) - get<Y>(last_target)) < abs((get<Y>(current_target) - horizontal_boundary * 2) - get<Y>(last_target))
+				&& abs(get<Y>(current_target) - get<Y>(last_target)) < abs((get<Y>(current_target) + horizontal_boundary * 2) - get<Y>(last_target))) {
+				get<Y>(new_target) += get<Y>(current_target) - get<Y>(last_target);
+			} else if (abs((get<Y>(current_target) - horizontal_boundary * 2) - get<Y>(last_target)) < abs((get<Y>(current_target) + horizontal_boundary * 2) - get<Y>(last_target))) {
+				get<Y>(new_target) += (get<Y>(current_target) - horizontal_boundary * 2) - get<Y>(last_target);
+			} else {
+				get<Y>(new_target) += (get<Y>(current_target) + horizontal_boundary * 2) - get<Y>(last_target);
+			}
+
+			return new_target;
+		};
+
+		auto wrap = [](double position, double min, double max) -> double {
+			if (position < min) {
+				return position + (max - min);
+			} else if (position > max) {
+				return position + (min - max);
+			} else {
+				return position;
+			}
+		};
+
+		auto wrap_trail = [=](Point &current, Point &target) -> void {
+			double new_x = wrap(get<X>(current), -horizontal_boundary, +horizontal_boundary);
+			get<X>(target) += new_x - get<X>(current);
+			get<X>(current) = new_x;
+
+			double new_y = wrap(get<Y>(current), -vertical_boundary, +vertical_boundary);
+			get<Y>(target) += new_y - get<Y>(current);
+			get<Y>(current) = new_y;
+		};
+
+		auto move_trail = [=](Point &current, Point &target) -> void {
+			wrap_trail(current, target);
+
+			get<X>(current) += (get<X>(target) - get<X>(current)) * TrailSprite::get_trail_space();
+			get<Y>(current) += (get<Y>(target) - get<Y>(current)) * TrailSprite::get_trail_space();
+		};
+
+		{
+			TrailSprite &trail = get_trail(TrailSprite::get_trail_length() - 1);
+			trail.target_position = get_target(Point(final<X>(), final<Y>()), trail.target_position);
+			move_trail(trail.home(), trail.target_position);
+			trail.change_texture(m_texture);
+		}
+		for (size_t i = TrailSprite::get_trail_length() - 1; i > 0; i--) {
+			TrailSprite &trail = get_trail(i - 1);
+			const TrailSprite &target_trail = get_trail(i);
+			trail.target_position = get_target(Point(target_trail.final<X>(), target_trail.final<Y>()), trail.target_position);
+			move_trail(trail.home(), trail.target_position);
+			trail.change_texture(m_texture);
```

Comment:
I uh
think that's basically how I was approaching this
though I'd calculate the ghost sprite's positions and use the closest one
The issue I had is that it would break if the trail sprite ended up lagging so far behind that the actual sprite would be closer than the ghost sprite and it'd end up suddenly changing directions